core/src/main/java/com/linecorp/armeria/client/HttpClientDelegate.java

```diff
@@ -62,12 +58,22 @@
     public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) throws Exception {
         final Endpoint endpoint = ctx.endpoint();
         if (endpoint == null) {
-            // Note that this response will be ignored because:
-            // - `ClientRequestContext.endpoint()` returns `null` only when the context initialization failed.
-            // - `ClientUtil.initContextAndExecuteWithFallback()` will use the fallback response rather than
-            //   what we return here.
-            req.abort(CONTEXT_INITIALIZATION_FAILED);
-            return HttpResponse.ofFailure(CONTEXT_INITIALIZATION_FAILED);
+            // It is possible that we reach here even when `EndpointGroup` is not empty,
+            // because `endpoint` can be `null` for the following two cases:
+            // - `EndpointGroup.select()` raised an `EmptyEndpointGroupException`.
+            // - Other exception was raised while context initialization.
+            //
+            // Because all the clean-up is done by `DefaultClientRequestContext.failEarly()`
+            // when context initialization fails with an exception other than `EmptyEndpointGroupException`,
+            // we can assume that the exception and response created here will be exposed only
+            // when context initialization failed due to an `EmptyEndpointGroupException`.
+            //
+            // See `DefaultClientRequestContext.init()` for more information.
+            System.err.println("!!!!!!");
```

Comment:
Is the write to stderr here on purpose or a leftover after a debugging session?

Reply:
!!!!!!